sdk/eventhub/azure-eventhubs/azure/eventhub/aio/eventprocessor/event_processor.py

```diff
@@ -0,0 +1,250 @@
+            from azure.eventhub.aio import EventHubClient
+            from azure.eventhub.eventprocessor import EventProcessor, PartitionProcessor, Sqlite3PartitionManager
+
+            class MyPartitionProcessor(object):
+                async def process_events(self, events, partition_context):
+                    if events:
+                        # do something sync or async to process the events
+                        await partition_context.update_checkpoint(events[-1].offset, events[-1].sequence_number)
+
+
+            client = EventHubClient.from_connection_string("<your connection string>", receive_timeout=5, retry_total=3)
+            partition_manager = Sqlite3PartitionManager()
+            try:
+                event_processor = EventProcessor(client, "$default", MyPartitionProcessor, partition_manager)
+                asyncio.ensure_future(event_processor.start())
+                await asyncio.sleep(100)  # allow it to run 100 seconds
+                await event_processor.stop()
+            finally:
+                await partition_manager.close()
+
+    """
+    def __init__(
+            self, eventhub_client: EventHubClient, consumer_group_name: str,
+            partition_processor_factory: Callable[..., PartitionProcessor],
+            partition_manager: PartitionManager, **kwargs
+    ):
+        """
+        Instantiate an EventProcessor.
+
+        :param eventhub_client: An instance of ~azure.eventhub.aio.EventClient object
+        :type eventhub_client: ~azure.eventhub.aio.EventClient
+        :param consumer_group_name: The name of the consumer group this event processor is associated with. Events will
+         be read only in the context of this group.
+        :type consumer_group_name: str
+        :param partition_processor_factory: A callable(type or function) object that creates an instance of a class
+         implementing the ~azure.eventhub.eventprocessor.PartitionProcessor.
+        :type partition_processor_factory: callable object
+        :param partition_manager: Interacts with the storage system, dealing with ownership and checkpoints.
+         For preview 2, sample Sqlite3PartitionManager is provided.
+        :type partition_manager: Class implementing the ~azure.eventhub.eventprocessor.PartitionManager.
+        :param initial_event_position: The offset to start a partition consumer if the partition has no checkpoint yet.
+        :type initial_event_position: int or str
+        :param polling_interval: The interval between any two pollings of balancing and claiming
+        :type float
+
+        """
+
+        self._consumer_group_name = consumer_group_name
+        self._eventhub_client = eventhub_client
+        self._eventhub_name = eventhub_client.eh_name
+        self._partition_processor_factory = partition_processor_factory
+        self._partition_manager = partition_manager
+        self._initial_event_position = kwargs.get("initial_event_position", "-1")
```

Comment:
For Python3 only code where you are directly consuming arguments (and not just passing them on to someone else), please use [keyword-only](https://www.python.org/dev/peps/pep-3102/) arguments with defaults rather than `**kwargs`.

Reply:
Changed to keyword only arguments